app/router.jsx

```diff
@@ -31,41 +32,44 @@ import SupportUs from './components/SupportUs'
-      <Route path="/login" component={LoginForm} />
-      <Route path="/logout" component={LogoutPage} />
-      <Route path="/login/callback/:provider" component={ThirdPartyCallback} />
-      <Route path="/confirm_email/:token" component={ConfirmEmail} />
-      <Route path="/reset_password" component={ResetPasswordRequestForm} />
-      <Route path="/reset_password/confirm/:token" component={ResetPasswordConfirmForm} />
-      <Route path="/newsletter/unsubscribe/:token" component={NewsletterSubscription} />
-      <Route path="/u/:username" component={User}>
-        <IndexRoute component={UserProfile} />
-        <Route path="/u/:username/activity" component={ActivityLog} />
-        <Route path="/u/:username/settings" component={UserSettings} />
-        <Route path="/u/:username/videos" component={UserAddedVideos} />
-        <Route path="/u/:username/subscriptions" component={SubscriptionsPage} />
-        <Route path="/u/:username/notifications" component={NotificationsPage} />
-      </Route>
-      <Route path="/videos" component={VideosIndexPage} />
-      <Route path="/videos/add" component={AddVideoForm} />
-      <Route path="/videos/add/:videoUrl" component={AddVideoForm} />
-      <Route path="/videos/:videoId" component={VideoDebate} view="debate" />
-      <Route path="/videos/:videoId/history" component={VideoDebate} view="history" />
-      <Route path="/s/:slug_or_id" component={SpeakerPage} />
-      <Route path="/help" component={Help} />
-      <Route path="/help/*" component={Help} />
-      <Route path="/extension" component={BrowserExtensionsPage} />
-      <Route path="/moderation" component={Moderation} />
-      <Route path="/support-us" component={SupportUs} />
-      <Route path="/search" component={SearchPage} />
-      <Route path="/search/:entity" component={SearchPage} />
-      <Route path="*" component={NotFound} />
-    </Route>
+  <Router history={createBrowserHistory}>
+    <Layout>
+      <Switch>
+        <Route path="/" exact component={Home} />
+        <Route path="/signup" exact component={SignupForm} />
+        <Route path="/signup/callback/:provider" exact component={ThirdPartyCallback} />
+        <Route path="/login" exact component={LoginForm} />
+        <Route path="/logout" exact component={LogoutPage} />
+        <Route path="/login/callback/:provider" exact component={ThirdPartyCallback} />
+        <Route path="/confirm_email/:token" exact component={ConfirmEmail} />
+        <Route path="/reset_password" exact component={ResetPasswordRequestForm} />
+        <Route path="/reset_password/confirm/:token" exact component={ResetPasswordConfirmForm} />
+        <Route path="/newsletter/unsubscribe/:token" exact component={NewsletterSubscription} />
+        <Route path="/u/:username">
+          <User>
+            <Switch>
+              <Route path="/u/:username" exact component={ActivityLog} />
+              <Route path="/u/:username/profile" exact component={UserProfile} />
```

Comment:
@francoispeyret I integrated your changes from https://github.com/CaptainFact/captain-fact-frontend/pull/976 directly here and in `app/components/Users/UserMenu.jsx`. I hope that's ok!